apm-agent-plugins/apm-opentelemetry/apm-opentelemetry-plugin/src/main/java/co/elastic/apm/agent/opentelemetry/sdk/BehavioralAttributes.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.opentelemetry.sdk;
+
+/**
+ * An extension to the OTel API that allows us to provide special tracing settings.
+ * Such settings may affect tracing behavior internally, but they are not added as data and not persisted as span attributes.
+ */
+public class BehavioralAttributes {
+
+    /**
+     * Can be used to make a span non-discardable,
+     * In some cases, spans may be discarded, for example if {@code span_min_duration} config option is set and the span does not exceed
+     * the configured threshold. Use this attribute to make sure the current span is not discarded.
+     *
+     * NOTE: making a span non-discardable implicitly makes the entire stack of active spans non-discardable as well. Child spans can
+     * still be discarded.
+     */
+    public static final String NON_DISCARDABLE = "elastic.otel.nondiscardable";
```

Comment:
Per the [OTel naming schema recommendations](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-naming.md#recommendations-for-application-developers), I suggest using `co.elastic` as a prefix.
Also, I'd prefer discardable=false rather than nondiscardable=true
```suggestion
    public static final String DISCARDABLE = "co.elastic.discardable";
```

Reply:
:+1: if you merge the suggestion make sure to also update/comment on https://github.com/elastic/apm/issues/645 so we keep a consistent proposal.